src/Mvc/Mvc.ApiExplorer/src/ApiResponseTypeProvider.cs

```diff
@@ -157,6 +179,18 @@ private static List<IApiResponseMetadataProvider> GetResponseMetadataAttributes(
                         }
                     }
 
+                    // We special case the handling of ProcuesResponseTypeAttributes since
+                    // multiple ProducesResponseTypeAttributes are permitted on a single
+                    // action/controller/etc. In that scenario, instead of picking the most-specific
+                    // set of content types (like we do with the Produces attribute above) we process
+                    // the content types for each attribute independently.
+                    if (metadataAttribute is ProducesResponseTypeAttribute)
```

Comment:
Hmm, does this cause issues if you have something like so:

```C#
[Produces("application/json")]
public class MyController
{
    [ProducesResponseType(200, typeof(Product)", "text/plain")]
    public IActionResult MyAction()
}
```

Reply:
Because we walk the attributes in order from least-specific to most-specific, and we end up maintaining a statusCode : ApiResponseType associated with them:

https://github.com/dotnet/aspnetcore/blob/213cf6dbc315891456ea9624e3d66d69dc03fdb1/src/Mvc/Mvc.ApiExplorer/src/ApiResponseTypeProvider.cs#L160-L163

In the scenario above, the expected output would look have a content-type of `text/plain` instead of `application/json` since that is the most specific attribute for that particular status code.

Same applies if the action had a `Produces("text/plain")`.

If you had:

```C#
[Produces("application/json")]
public class MyController
{
    [ProducesResponseType(200, typeof(Product)", "text/plain")]
    [ProducesResponseType(200, typeof(Product)", "text/xml")]
    public IActionResult MyAction()
}
```

Then `text/xml` would be the resulting format type.

Reply:
Dope. Would you mind adding this as a unit test? I don't think we need multiple ProducesResponseType for the same status code, but just to make sure controller values get overriden by the action.

Reply:
Yep -- we can tack this on to the `ApiExplorerResponseTypeOverrideOnActionController` scenarios.